### PR TITLE
feature/refresh-logs-on-try-it

### PR DESCRIPTION
**Pull request by gratcliff**
Functionality to refresh logs when user clicks the 'try it now' button in our explorer.

### CODE REVIEW

**Comment by domharrington** [comment]
I'm looking at this now! We don't need to add everyone as a reviewer for a feature like this - 1 or 2 people is fine.

File: packages/api-explorer/src/Doc.jsx [hunk 1]
```diff
@@ -256,6 +264,8 @@ class Doc extends React.Component {
           url,
           method,
         }}
+        tryItRequestFired={this.state.tryItRequestFired}
+        onReset={this.resetTryItRequest()}
```
[the remark this answers is not in the document]
Comment (gratcliff) [reply packages/api-explorer/src/Doc.jsx]: I'm not sure why any of this happened, I blocked the recursive calls through passing an update to the parent's state. Also, log requests worked repeatedly.

**Comment by domharrington** [comment]
@gratcliff can you test this? I simplified the prop passing a little bit and also simplified the way the group is emitted to the other logs components by storing it at the top level. It appears to work for me but I may have missed something.

**Comment by gratcliff** [comment]
@domharrington let's go with your suggestion and get this running in a PR app. Everything seems to work fine. Tried to replicate lag times locally, couldn't catch any issues.

**Comment by dok** [comment]
Hey @gratcliff, do you have the PR to the main repo? I would also like to see this in action on the main application.

**Comment by gratcliff** [comment]
> Hey @gratcliff, do you have the PR to the main repo? I would also like to see this in action on the main application.

Hey @dok, there isn't one up yet, as this hasn't been merged/tagged for release.

**Comment by domharrington** [comment]
As tested over in the readme pr app, we're having some issues around when to make the request to fetch new logs, since the log insertion happens asynchronously to our metrics backend. Lets use this to add retries when fetching new logs: https://github.com/zeit/async-retry.

I guess we can compare the number of logs returned against the current number of logs, and if it's the same, then we retry,

**Pull request merged**